x-pack/plugins/ml/public/jobs/new_job/wizard/steps/index_or_search/index_or_search_controller.js

```diff
@@ -48,7 +48,7 @@ module.controller('MlNewJobStepIndexOrSearch',
     timefilter.disableTimeRangeSelector(); // remove time picker from top of page
     timefilter.disableAutoRefreshSelector(); // remove time picker from top of page
 
-    $scope.indexPatterns = getIndexPatterns();
+    $scope.indexPatterns = getIndexPatterns().filter(indexPattern => !indexPattern.get('type'));
```

Comment:
@peteharverson `type` is a new field on index patterns, so rollup ones are the only ones that will have a value. I don't foresee any other types in the near future. This way of filtering is to avoid rollup references in OSS code, but I suppose it is different here since ML is x-pack 😃Thanks for the constants file path. I'll think on this a little more, including our policies around referencing x-pack plugins from another x-pack plugin, and may adjust this filtering in my next pass.

Reply:
I was going to bring this up later, but I would like to reopen the discussion around avoiding referencing rollup index patterns from within OSS Kibana (and from within X-Pack plugins by extension).

I think that it makes sense to avoid referencing X-Pack features from within OSS (e.g. monitoring, reporting), because those things literally cannot exist unless X-Pack is installed. An OSS installation of the stack will never encounter those things. But rollups are different, because they generate artifacts in the form of rollup indices, which _can_ exist within OSS installations. I think it makes sense for OSS Kibana to have awareness of these artifacts and the capability of dealing with them intelligently. So I think this is one very good case for making an exception to the rule.